Persistence/Repositories/TransactionRepository.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Domain;
+
+using Microsoft.EntityFrameworkCore;
+
+namespace Persistence.Repositories
+{
+    public class TransactionRepository : ITransactionRepository
+    {
+        private readonly HaSpManContext _haSpManContext;
+
+        public TransactionRepository(HaSpManContext haSpManContext)
+        {
+            _haSpManContext = haSpManContext;
+        }
+        public async Task<Transaction> GetByIdAsync(Guid id)
+        {
+
+            return await _haSpManContext.Transactions
+                .FirstOrDefaultAsync(x => x.Id == id);
+        }
+
+
+
+        public async Task<IEnumerable<Transaction>> GetAllAsync()
+        {
+            return await _haSpManContext.Transactions.ToListAsync();
+        }
+
+        public async Task<int> GetLastTransactionSequence()
```

Comment:
This would be subject to race condition collisions, I think. It would be better to write a customer class that inherits from the [`ValueGenerator` class](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.valuegeneration.valuegenerator?view=efcore-5.0), that way, values would be generated when they are saved to the database.